src/controls/scale.ts

```diff
@@ -70,6 +70,13 @@ export function scalingIsForbidden(
   if (lockY && by === 'y') {
     return true;
   }
+  // object disappears if scaling 0 sized object so forbid
+  if (fabricObject.width === 0 && by == 'x') {
+    return true;
+  }
+  if (fabricObject.height === 0 && by === 'y') {
+    return true;
+  }
```

Comment:
```suggestion
  if (fabricObject.width === 0 && by !== 'y') {
    return true;
  }
  if (fabricObject.height === 0 && by !== 'x') {
    return true;
  }
```

Reply:
I tried this with a 0 height rectangle. It turned out that the bottom right corner is what was showing on the right side, so I couldn't extend the rectangle in the x direction. That said, I found out the diagonal was working because it used uniform scaling and the uniform scaling calculation uses a different calculation based on adding the scales and doesn't break as a result. But if the uniform scaling was off, the diagonal ends up with the issue (i.e., NaN and the polygon disappears).

So this is a good thought since non-uniform scaling will break without taking the 'both' case into consideration. But it almost begs for the diagonals to be hidden. Is there an easy way to do this (or would that be a bad idea since it could add confusion on how corners show)?

Reply:
Not sure I understand

Reply:
Basically, because it’s a line, the diagonal scalers stack on top of the edge scaler and making the both case forbidden means you can’t extend the line.

To make things confusing, when uniform scaling is on and both is allowed, the diagonals work but when uniform scaling is off, this diagonals are broken.

Reply:
i m pretty much sure there was code for the simple Line object that was hiding the corners when one dimension was 0.

Reply:
Regarding @ShaMan123 comment let's take the most restrictive fix, so if one case of diagonals would work, we don't care we lock both.
As a side effect a rectangle created with zero width won't even stretch, it won't error, but in general developer ahve to be driven away from such cases.

Reply:
we could even argue that during the render cycle an object with width/height of 0 is not on screen and should therefore be skipped and not rendered at all

Reply:
seems I have found the source of these bugs


<img width="1512" alt="Screenshot 2024-01-06 at 15 07 26" src="https://github.com/fabricjs/fabric.js/assets/34343793/738f685f-426d-44eb-874f-efdb876b7985">

Should be `!width || !height`

Reply:
```
  isNotVisible() {
    return !this.opacity || !this.width || !this.height || !this.visible;
  }

```

Reply:
IMO it is weird the an object with height of 0 and strokeWidth >0 draws the stroke

Reply:
that only skips rendering, it doesn't makes it unselectable. If you have padding the object is still selectable.

Reply:
> seems I have found the source of these bugs
> 
> <img alt="Screenshot 2024-01-06 at 15 07 26" width="1512" src="https://private-user-images.githubusercontent.com/34343793/294685422-738f685f-426d-44eb-874f-efdb876b7985.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NTI0MDMsIm5iZiI6MTcwNDU1MjEwMywicGF0aCI6Ii8zNDM0Mzc5My8yOTQ2ODU0MjItNzM4ZjY4NWYtNDI2ZC00NGViLTg3NGYtZWZkYjg3NmI3OTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA2VDE0NDE0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyNWRkM2VjYzNiMmM3Mjc5ZjRhZDQ0YTlkZTc5NTQ0NjkyZjFlNTBmNWZkMjJlNzNlYzA2ZjU1YmJmZjc2OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rxZISNomW-VEhX5A1K0r3orcuEVgBt6r8yKsOJHRnlw">
> Should be `!width || !height`

This is unrelated.
the boolean visible is for the developer, and determines rendering and selectability.
The 'isNotVisible' is an internal check to skip rendering code when will produce no output. It has nothing to do with our calculation blowing away because of a zero division.